app/components/AccountSetup.vue

```diff
@@ -1,55 +1,72 @@
 <script setup lang="ts">
-import {  NCollapseItem, NIcon, NButton, NSpin } from 'naive-ui'
-import { computed, ref } from 'vue'
-import type {Ref} from 'vue'
-import { Wallet as WalletIcon } from '@vicons/ionicons5'
-import { Wallet } from 'ethers'
+import { NCollapseItem, NIcon, NButton, NSpin } from 'naive-ui';
+import { computed, ref } from 'vue';
+import type { Ref } from 'vue';
+import { Wallet as WalletIcon } from '@vicons/ionicons5';
+import { Wallet } from 'ethers';
 
-const emits = defineEmits<{(e: 'getCreatedWallet', accountAddress: string, accountPrivateKey: string, accountGenerationDate: Date): void}>()
-const publicAddress: Ref<string | null> = ref(null)
-const isCreating = ref(false)
+const emits = defineEmits<{
+  (
+    e: 'getCreatedWallet',
+    accountAddress: string,
+    accountPrivateKey: string,
+    accountGenerationDate: Date
+  ): void;
+}>();
+const publicAddress: Ref<string | null> = ref(null);
+const isCreating = ref(false);
 
-const title = computed(() => publicAddress.value && !isCreating.value
-? `Created wallet ${publicAddress.value}`
-: 'Create new or restore existing wallet')
+const title = computed(() =>
+  publicAddress.value && !isCreating.value
+    ? `Created wallet ${publicAddress.value}`
+    : 'Create new or restore existing wallet'
+);
 
 const generateWallet = async () => {
-    const wallet = Wallet.createRandom()
-    publicAddress.value = wallet.address
-    isCreating.value = true
-    await setTimeout(() => {
-        isCreating.value = false
-    }, 3000)
-    emits('getCreatedWallet', wallet.address, wallet.privateKey, new Date())
-}
+  const wallet = Wallet.createRandom();
+  publicAddress.value = wallet.address;
+  isCreating.value = true;
+
+  await new Promise(() => {
+    setTimeout(() => {
+      isCreating.value = false;
+    }, 3000);
+    emits('getCreatedWallet', wallet.address, wallet.privateKey, new Date());
+  });
```

Comment:
I think the emit should happen after the `await`ed promise (because emit will trigger the collapse animation later on)